src/main/java/com/dabsquared/gitlabjenkins/gitlab/api/impl/ResteasyGitLabClientBuilder.java

```diff
@@ -90,19 +90,22 @@ public final GitLabClient buildClient(String url, String apiToken, boolean ignor
 
     private GitLabClient buildClient(String url, String apiToken, ProxyConfiguration httpProxyConfig, boolean ignoreCertificateErrors, int connectionTimeout, int readTimeout) {
         ResteasyClientBuilder builder = new ResteasyClientBuilder();
+
         if (ignoreCertificateErrors) {
             builder.hostnameVerification(ResteasyClientBuilder.HostnameVerificationPolicy.ANY);
             builder.disableTrustManager();
         }
 
-        Proxy proxy = httpProxyConfig != null ? httpProxyConfig.createProxy(getHost(url)) : NO_PROXY;
-        if (proxy != NO_PROXY) {
-            InetSocketAddress address = (InetSocketAddress) proxy.address();
-            builder.defaultProxy(address.getHostName().replaceFirst("^.*://", ""),
-                address.getPort(),
-                address.getHostName().startsWith("https") ? "https" : "http",
-                httpProxyConfig.getUserName(),
-                httpProxyConfig.getPassword());
+        if (httpProxyConfig != null) {
+            Proxy proxy = httpProxyConfig.createProxy(getHost(url));
+            if (proxy.type() == HTTP) {
```

Comment:
This change ignores socks-proxy-setups as we currently force the proxy's protocol to http(s) anyway. Does anyone know whether the plugin ever worked with socks-proxys?

Reply:
I don't know if it ever did.

Reply:
Me either.